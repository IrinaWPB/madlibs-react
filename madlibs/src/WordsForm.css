.WordsForm {
    width: 400px;
    margin-left: auto;
    margin-right: auto;
   
}

input {
    width: 300px;
    margin-bottom: 10px;
}
