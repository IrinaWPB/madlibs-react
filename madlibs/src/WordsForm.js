import React, {useState, useEffect} from 'react'
import './WordsForm.css'

/* Form collects all the words and passes data object to the parent component*/

const WordsForm = ({ getWords }) => {
    const INITIAL_STATE = {
        name:"",
        adjective:"",
        noun1:"",
        verb:"",
        noun2:""
    }
    const [data, setData] = useState(INITIAL_STATE)
    const [disabled, setDisabled] = useState(true)

    //Checks if all the fields are filled to enable the submit button 
    useEffect(() => {
        if (Object.values(data).indexOf("") === -1) {
            setDisabled(false)
        } else setDisabled(true)
    }, [data])

    const handleChange = e => {
        const { name, value } = e.target
        setData(data => ({
            ...data,
            [name]: value
        }))
        
    }

    const handleSubmit = e => {
        e.preventDefault()
        getWords(data)
        setData(INITIAL_STATE)
    }

    return (
        <form onSubmit={handleSubmit} className="WordsForm">
            <label htmlFor="name"></label>
            <input id="name" type="text" 
                   placeholder="name" name="name"
                   value={data.name} onChange={handleChange}></input>
            <label htmlFor="adjective"></label>
            <input id="adjective" type="text" 
                   placeholder="adjective" name="adjective"
                   value={data.adjective} onChange={handleChange}></input>
            <label htmlFor="noun1"></label>
            <input id="noun1" type="noun1" 
                   placeholder="noun 1" name="noun1"
                   value={data.noun1} onChange={handleChange}></input>
            <label htmlFor="verb"></label>
            <input id="verb" type="verb" 
                   placeholder="verb" name="verb"
                   value={data.verb} onChange={handleChange}></input>
            <label htmlFor="noun2"></label>
            <input id="noun2" type="text" 
                   placeholder="noun 2" name="noun2"
                   value={data.noun2} onChange={handleChange}></input>
            <button disabled={disabled}>Get a Story</button>
        </form>
    ) 
}

export default WordsForm