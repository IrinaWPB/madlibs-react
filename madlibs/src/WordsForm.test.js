import React from 'react'
import { render } from '@testing-library/react'
import WordsForm from './WordsForm'

it('renders without crashing', () => {
    render(<WordsForm />)
})

it('matches the snapshot', () => {
    const {asFragment} = render(<WordsForm />)
    expect(asFragment()).toMatchSnapshot()
})