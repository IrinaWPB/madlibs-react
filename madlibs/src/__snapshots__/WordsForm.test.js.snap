// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="WordsForm"
  >
    <label
      for="name"
    />
    <input
      id="name"
      name="name"
      placeholder="name"
      type="text"
      value=""
    />
    <label
      for="adjective"
    />
    <input
      id="adjective"
      name="adjective"
      placeholder="adjective"
      type="text"
      value=""
    />
    <label
      for="noun1"
    />
    <input
      id="noun1"
      name="noun1"
      placeholder="noun 1"
      type="noun1"
      value=""
    />
    <label
      for="verb"
    />
    <input
      id="verb"
      name="verb"
      placeholder="verb"
      type="verb"
      value=""
    />
    <label
      for="noun2"
    />
    <input
      id="noun2"
      name="noun2"
      placeholder="noun 2"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Get a Story
    </button>
  </form>
</DocumentFragment>
`;
